ably/proto_message_decoding_test.go

```diff
@@ -0,0 +1,147 @@
+package ably
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/ably/ably-go/ably/internal/ablyutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type fixture struct {
+	Data          string `json:"data"`
+	Encoding      string `json:"encoding"`
+	ExpectedType  string `json:"expectedType"`
+	ExpectedValue any    `json:"expectedValue"`
+}
+
+func loadFixtures() ([]fixture, error) {
+	var dec struct {
+		Messages []fixture
+	}
+
+	// We can't embed the fixtures as go:embed forbids embedding of resources above the current directory.
+	text, err := os.ReadFile("../common/test-resources/messages-encoding.json")
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(text, &dec)
+	return dec.Messages, err
+}
+
+func Test_decodeMessage(t *testing.T) {
+	fixtures, err := loadFixtures()
+	require.NoError(t, err, "failed to load test fixtures")
+	for _, f := range fixtures {
+		t.Run(f.Data, func(t *testing.T) {
```

Comment:
Is it worth giving each fixture a human readable name rather than using the data as the test name?

Reply:
That is what I did in the msgpack fixtures.
But the messages-encoding.json file was created by [many years ago](https://github.com/ably/ably-common/pull/17) and does not have a name field. The fixtures in this file were not used in any tests in ably-go, so I decided to add one. But I did not intend to alter this file, which probably is imported by other SDKs.

Reply:
Ok I was implying we add a name to the fixture file, but I'll leave it to your judgement, happy to also leave as is.

Reply:
I don't really want to change  messages-encoding.json as it is being imported by other SDKs

Reply:
👍